File: Leetcode/Daily/04-06-2024/prob.cpp
class Solution {
public:
    int longestPalindrome(string s) {
        map<char,int>mp;
        for(auto &c:s){
            mp[c]++;
        }
        int odd = 0;
        int len = 0;
        for(auto &c:mp){
            if(c.second&1){
                if(odd==0){
                    odd = 1;
                    len+= c.second;
                }
                else{
                    len += c.second-1;
                }
            }
            else len += c.second;
        }

        return len;
    }
};

File: Leetcode/Daily/04-06-2024/prob.go
func longestPalindrome(s string) int {
    mp := make(map[rune]int)

    for _,c := range s{
        mp[c]++
    }
    cnt := 0
    odd := 0
    for _,k := range mp{
        if k%2==0{
            cnt += k
        }else{
          
            cnt += k-1
            odd = 1
            
        }
    }
    if odd==1{
        cnt++
    }
    return cnt
}